app/controllers/channel_transfer_controller.rb

```diff
@@ -0,0 +1,37 @@
+class ChannelTransferController < ApplicationController
+  include ChannelsHelper
+
+  before_action :verify_token
+
+  attr_reader :current_channel
+
+  def reject_transfer
+    Channels::RejectChannelTransfer.new(channel: current_channel).perform
+    redirect_to home_publishers_path, notice: t("shared.channel_transfer_rejected")
+  end
+
+  private
+
+  def verify_token
+    @current_channel = Channel.find(params[:id])
+    if @current_channel.nil? || @current_channel.contest_token.empty? || @current_channel.contest_token != params[:token_id]
+      respond_to do |format|
+        format.json {
+          head 404
```

Comment:
It might be easier if we added a constraint in the `routes.rb` file
`get :reject_transfer, to: 'channel_transfer#reject_transfer' constraints: lambda { |req| req.format == :json }`

Then we wouldn't have to specify the format and have the duplicated logic